Source/Diagnostics/BoostedFrameDiagnostic.cpp

```diff
@@ -1374,40 +1370,56 @@ AddPartDataToParticleBuffer(
-                        tmp_particle_buffer[isp].GetRealData(DiagIdx::x).begin(),
-                        tmp_particle_buffer[isp].GetRealData(DiagIdx::x).end());
-
-        particles_buffer_[isp].GetRealData(DiagIdx::y).insert(
-                        particles_buffer_[isp].GetRealData(DiagIdx::y).end(),
-                        tmp_particle_buffer[isp].GetRealData(DiagIdx::y).begin(),
-                        tmp_particle_buffer[isp].GetRealData(DiagIdx::y).end());
-
-        particles_buffer_[isp].GetRealData(DiagIdx::z).insert(
-                        particles_buffer_[isp].GetRealData(DiagIdx::z).end(),
-                        tmp_particle_buffer[isp].GetRealData(DiagIdx::z).begin(),
-                        tmp_particle_buffer[isp].GetRealData(DiagIdx::z).end());
-
-        particles_buffer_[isp].GetRealData(DiagIdx::ux).insert(
-                        particles_buffer_[isp].GetRealData(DiagIdx::ux).end(),
-                        tmp_particle_buffer[isp].GetRealData(DiagIdx::ux).begin(),
-                        tmp_particle_buffer[isp].GetRealData(DiagIdx::ux).end());
-
-        particles_buffer_[isp].GetRealData(DiagIdx::uy).insert(
-                        particles_buffer_[isp].GetRealData(DiagIdx::uy).end(),
-                        tmp_particle_buffer[isp].GetRealData(DiagIdx::uy).begin(),
-                        tmp_particle_buffer[isp].GetRealData(DiagIdx::uy).end());
-
-        particles_buffer_[isp].GetRealData(DiagIdx::uz).insert(
-                        particles_buffer_[isp].GetRealData(DiagIdx::uz).end(),
-                        tmp_particle_buffer[isp].GetRealData(DiagIdx::uz).begin(),
-                        tmp_particle_buffer[isp].GetRealData(DiagIdx::uz).end());
+        // allocate size of particle buffer array to np
+        // This is a growing array. Each time we add np elements
+        // to the existing array which has size = init_size
+        int init_size = particles_buffer_[isp].GetRealData(DiagIdx::w).size();
+        int total_size = init_size + np;
+        particles_buffer_[isp].resize(total_size);
+
+        // Data pointers to particle attributes //
+        Real* const AMREX_RESTRICT wp_buff =
+              particles_buffer_[isp].GetRealData(DiagIdx::w).data();
+        Real* const AMREX_RESTRICT x_buff =
+              particles_buffer_[isp].GetRealData(DiagIdx::x).data();
+        Real* const AMREX_RESTRICT y_buff =
+              particles_buffer_[isp].GetRealData(DiagIdx::y).data();
+        Real* const AMREX_RESTRICT z_buff =
+              particles_buffer_[isp].GetRealData(DiagIdx::z).data();
+        Real* const AMREX_RESTRICT ux_buff =
+              particles_buffer_[isp].GetRealData(DiagIdx::ux).data();
+        Real* const AMREX_RESTRICT uy_buff =
+              particles_buffer_[isp].GetRealData(DiagIdx::uy).data();
+        Real* const AMREX_RESTRICT uz_buff =
+              particles_buffer_[isp].GetRealData(DiagIdx::uz).data();
+
+        Real* const AMREX_RESTRICT wp_temp =
```

Comment:
Couldn't all of the `_temp` be `Real const * const` (e.g., const pointer to const elements) instead of `Real * const` (e.g., const pointer to modifiable elements)?

Reply:
Oh yes. I have made this change for all _temp arrays